src/Sparrow/Json/AbstractBlittableJsonTextWriter.cs

```diff
@@ -502,20 +502,33 @@ private void WriteRawString(byte* buffer, int size)
 
         private void UnlikelyWriteLargeRawString(ReadOnlySpan<byte> buffer)
         {
+            ref byte bufferStart = ref MemoryMarshal.GetReference(buffer);
+            ref byte destination = ref Unsafe.AsRef<byte>(_buffer);
+
             // need to do this in pieces
             var posInStr = 0;
             while (posInStr < buffer.Length)
             {
-                var amountToCopy = Math.Min(buffer.Length - posInStr, JsonOperationContext.MemoryBuffer.DefaultSize);
-                FlushInternal();
+                var amountToCopy = Math.Min(buffer.Length - posInStr, JsonOperationContext.MemoryBuffer.DefaultSize - _pos);
```

Comment:
Not really. We never allow to do this on bigger than `JsonOperationContext.MemoryBuffer.DefaultSize` therefore there is no condition where it could happen.